core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts

```diff
@@ -50,45 +55,62 @@ import { RelationshipFieldStore } from './store/relationship-field.store';
      */
     readonly #dotMessageService = inject(DotMessageService);
 
     /**
-     * A signal that controls the visibility of the existing content dialog.
-     * When true, the dialog is shown allowing users to select existing content.
-     * When false, the dialog is hidden.
+     * A readonly private field that holds a reference to the `DestroyRef` service.
+     * This service is injected into the component to manage the destruction lifecycle.
      */
-    $showExistingContentDialog = signal(false);
+    readonly #destroyRef = inject(DestroyRef);
 
     /**
-     * A signal that holds the menu items for the relationship field.
-     * These items control the visibility of the existing content dialog and the creation of new content.
+     * A readonly private field that holds an instance of the DialogService.
+     * This service is injected using Angular's dependency injection mechanism.
+     * It is used to manage dialog interactions within the component.
      */
-    $menuItems = signal<MenuItem[]>([
-        {
-            label: this.#dotMessageService.get(
-                'dot.file.relationship.field.table.existing.content'
-            ),
-            command: () => {
-                this.$showExistingContentDialog.update((value) => !value);
-            }
-        },
-        {
-            label: this.#dotMessageService.get('dot.file.relationship.field.table.new.content'),
-            command: () => {
-                // TODO: Implement new content
-            }
-        }
-    ]);
+    readonly #dialogService = inject(DialogService);
 
     /**
-     * A readonly instance of the RelationshipFieldStore injected into the component.
-     * This store is used to manage the state and actions related to the relationship field.
+     * Reference to the dynamic dialog. It can be null if no dialog is currently open.
+     *
+     * @type {DynamicDialogRef | null}
      */
-    readonly store = inject(RelationshipFieldStore);
+    #dialogRef: DynamicDialogRef | null = null;
+
+    /**
+     * A signal that holds the menu items for the relationship field.
+     * These items control the visibility of the existing content dialog and the creation of new content.
+     */
+    $menuItems = computed<MenuItem[]>(() => {
+        const isDisabledCreateNewContent = this.store.isDisabledCreateNewContent();
+
+        return [
+            {
+                label: this.#dotMessageService.get(
+                    'dot.file.relationship.field.table.existing.content'
+                ),
+                disabled: isDisabledCreateNewContent,
+                command: () => {
+                    this.showExistingContentDialog();
+                }
+            },
+            {
+                label: this.#dotMessageService.get('dot.file.relationship.field.table.new.content'),
+                command: () => {
+                    // TODO: Implement new content
```

Comment:
Will this be in another ticket?

Reply:
Yeah, this command will tackle in another card.